### PR TITLE
adding model code

### PR DESCRIPTION
**Pull request by pkasela**

### CODE REVIEW

File: deep_src/dataloader/dataloader.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        query_id = self.data_ids[idx]
+        query_data = self.data[query_id]
+        query_text = query_data['text']
+        
+        # pos document
+        pos_sample = random.choice(query_data['relevant_docs'])
+        pos_text = self.corpus[pos_sample]
+        
+        # neg document
+        neg_sample = random.choice(query_data['bm25_docs'])
```
Comment (ospitia) [inline deep_src/dataloader/dataloader.py:24]: are relevant_docs and bm25_docs disjoint?

File: deep_src/dataloader/utils.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        for lne in f:
+            yield json.loads(lne)
+
+def load_query_data(file: str, verbose: bool=True):
+    with open(file, 'r') as f:
+        query_file = {}
+        pbar = tqdm.tqdm(f, desc='Creating data for loading') if verbose else f
+        for lne in pbar:
+            query_json = json.loads(lne)
+            bm_25_not_relevant_docs = [id_ for id_ in query_json['bm25_doc_ids'] if id_ not in query_json['rel_doc_ids']]
```
Comment (ospitia) [inline deep_src/dataloader/utils.py:35]: ok, I see that here

File: deep_src/model/model.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        loss = relu(
+            self.margin - positive_embedding_scores + negative_embedding_scores
+        ).mean()
+
+        return loss
+
+class BiEncoder(nn.Module):
+    def __init__(self, query_model, doc_model, tokenizer, device):
+        super(BiEncoder, self).__init__()
+        self.query_model = query_model.to(device)
```
Comment (ospitia) [inline deep_src/model/model.py:33]: are you going to change this for a single model?

Comment (pkasela) [reply deep_src/model/model.py:33]: No, I have created a new class for the single shared model, I will push it later

**Pull request closed without merging**